src/ranges.rs

```diff
@@ -0,0 +1,77 @@
+use core::marker::PhantomData;
+use num;
+
+use typenum::{Unsigned, Cmp, B1, Same};
+// Although this says private, it's needed to write generic inequality
+// constraints for typenums
+use typenum::private::IsLessOrEqualPrivate;
+
+/// Indicates that a type-level number may be converted to a runtime-level
+/// number of the type T
+pub trait ReifyTo<T> {
+    fn reify() -> T;
+}
+
+impl<T: Unsigned> ReifyTo<u8> for T {
+    fn reify() -> u8 {
+        <T as Unsigned>::to_u8()
+    }
+}
+
+impl<T: Unsigned> ReifyTo<u16> for T {
+    fn reify() -> u16 {
+        <T as Unsigned>::to_u16()
+    }
+}
+
+#[derive(Debug)]
+pub struct Bounded<T,L,U> {
+    pub val: T,
```

Comment:
Doesn't the fact that this field is `pub` mean that it can be directly mutated to any value of type `T` by an external actor with mutable access to the `Bounded` instance?  Seems like a read-only getter (and likely a constructor) would be appropriate here to prevent that possibility.

Reply:
oh good grief, yes of course that's true. I'll add two setters, one checking and the other proof-carrying.

Reply:
Or maybe I can just leave out the setters for now, until they're needed. At the moment all ranges are constructed via the `clamp` function. (I have a constructor, I just took it out to keep this minimal for now)

Reply:
I think something like a constructor would be necessary for me to feel good about using this thing with the level of confidence I was expecting -- namely, I want to be able to use instances without ever worrying about if they are in an invalid state (and instead just worry about getting things transformed into that type)